$NetBSD$

Do not add -W* flags to CFLAGS.
NetBSD has not got strtonum(3) and strnvis(3) so use the compat/
alternatives.
Do not use rough install(1) commands, use cleaner INSTALL_* ones.

--- Makefile.orig	2009-01-17 21:12:07.000000000 +0100
+++ Makefile
@@ -46,7 +46,7 @@ SRCS= tmux.c server.c server-msg.c serve
       tty.c tty-term.c tty-keys.c tty-write.c
 
 CC?= cc
-INCDIRS+= -I. -I- -I/usr/local/include
+INCDIRS+=-I. -Icompat
 CFLAGS+= -DMETA="'${META}'"
 .ifdef PROFILE
 # Don't use ccache
@@ -61,10 +61,6 @@ CFLAGS+= -DBUILD="\"$(VERSION) ($(DATE))
 CFLAGS+= -DBUILD="\"$(VERSION)\""
 .endif
 #CFLAGS+= -pedantic -std=c99
-CFLAGS+= -Wno-long-long -Wall -W -Wnested-externs -Wformat=2
-CFLAGS+= -Wmissing-prototypes -Wstrict-prototypes -Wmissing-declarations
-CFLAGS+= -Wwrite-strings -Wshadow -Wpointer-arith -Wcast-qual -Wsign-compare
-CFLAGS+= -Wundef -Wbad-function-cast -Winline -Wcast-align
 
 PREFIX?= /usr/local
 INSTALLDIR= install -d
@@ -80,6 +76,14 @@ LIBS+= -lutil -lncurses
 # FreeBSD and DragonFly
 .if ${OS} == "FreeBSD" || ${OS} == "DragonFly"
 CFLAGS+= -DUSE_LIBUTIL_H
+LIBS+= -lcrypt
+.endif
+
+# NetBSD
+.if ${OS} == "NetBSD"
+SRCS+= compat/strtonum.c compat/vis.c
+LIBS+= -lcrypt
+CFLAGS+=-DNO_STRTONUM
 .endif
 
 OBJS= ${SRCS:S/.c/.o/:S/.y/.o/}
@@ -128,7 +132,7 @@ update-index.html:
 		sed "s/%%VERSION%%/${VERSION}/g" index.html.in >index.html
 
 install:	all
-		${INSTALLDIR} ${DESTDIR}${PREFIX}/bin
-		${INSTALLBIN} ${PROG} ${DESTDIR}${PREFIX}/bin/${PROG}
-		${INSTALLDIR} ${DESTDIR}${PREFIX}/man/man1
-		${INSTALLMAN} ${PROG}.1 ${DESTDIR}${PREFIX}/man/man1/
+		${INSTALL_PROGRAM_DIR} ${DESTDIR}${PREFIX}/bin
+		${INSTALL_PROGRAM} ${PROG} ${DESTDIR}${PREFIX}/bin/${PROG}
+		${INSTALL_MAN_DIR} ${DESTDIR}${PREFIX}/${PKGMANDIR}/man1
+		${INSTALL_MAN} ${PROG}.1 ${DESTDIR}${PREFIX}/${PKGMANDIR}/man1/
